consumer/consumererror/partial.go

```diff
@@ -0,0 +1,24 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+var noDataCount = -1
+
+type Partial struct {
```

Comment:
You already need `NewPartial` to meaningfully instantiate this, since `Partial` has unexported fields with no setters.

If we wanted to unexport it, we would need to make a `Partial` interface that the unexported `partial` struct implements and include an `errors.As` wrapper to handle this.

Reply:
Okay, makes sense. I don't think we gain much by changing this so I am going to mark this as resolved